bootspec/src/generation.rs

```diff
@@ -18,11 +18,24 @@ pub enum Generation {
     V1(v1::GenerationV1),
 }
 
+impl Generation {
+    /// The version of the bootspec document.
+    pub fn version(&self) -> u64 {
+        use Generation::*;
+
+        match self {
+            V1(_) => v1::SCHEMA_VERSION,
+        }
+    }
+}
+
 impl Serialize for Generation {
     fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
     where
         S: serde::Serializer,
     {
+        use Generation::*;
```

Comment:
nit: I think this would be better as explicit imports, but I think I understand why it's a wildcard and I think it's fine how it is.